src/router/Router.ts

```diff
@@ -40,15 +40,19 @@ export class Router {
 
     this.stackSize = stackSize;
     this.mapUrl = mapUrl;
-
   }
+
+  setMapUrl = (mapUrl: UrlMapper) => {
+    if (this.mapUrl) return;
```

Comment:
throw an error here, we don't want to silently ignore the setter. Also add initialUrl function to run the url mapper first time.

Reply:
@syaau  do you mean the function will look like this??
```
  setMapUrl = (mapUrl: UrlMapper, initialUrl?: string) => {
    if (this.mapUrl) throw new Error('Url mapper has already been declared');
    this.mapUrl = mapUrl;
    if (initialUrl) {
      this.setUrl(initialUrl);
    }
  };

```

Reply:
Yes, initialUrl needs to be set as this will be the main router initialization. The default initialUrl could be `/`.